### PR TITLE
DR-98 docstrings and docs repo

### PR DESCRIPTION
**Pull request by daveatweaviate**
[issue](https://semi-technology.atlassian.net/browse/DR-98)

Updates a couple of the error message links
Updates some of the outdated docs pages on the repo

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
 python3 -m venv .venv
 source .venv/bin/activate
 pip install -r requirements-devel.txt
 ```
 
-The next time you open your shell, you can activate your virtual environment using `source .venv/bin/activate`
+To activate your virtual environment, run `source .venv/bin/activate`.
 
-To run local instance of Weaviate, we recommend using Docker (see https://weaviate.io/developers/weaviate/installation/docker-compose).
+The newest client versions sometimes require upcoming Weaviate core features. We recommend using Docker (see https://weaviate.io/developers/weaviate/installation/docker-compose) to run a local instance of the latest Weaviate core for client development.
```
Comment (databyjp) [inline CONTRIBUTING.md:15]: Upcoming sounds unreleased

Comment (daveatweaviate) [reply CONTRIBUTING.md:15]: Yes, the core and client have both been developing ahead of the released server core.

File: weaviate/error_msgs.py [hunk 2]
```diff
@@ -14,9 +14,8 @@
 BATCH_MANUAL_USE_W = (
     "Manually batching means this code is NOT using the client's built-in "
     "multi-threading. To enable built-in multi-threading, set `batch_size` in "
-    " `client.batch.configure()` to an integer value . See also:"
-    "https://weaviate.io/developers/weaviate/current/restful-api-references/"
-    "batch.html#example-request-1"
+    " `client.batch.configure()` to an integer value . See also: "
+    "https://weaviate.io/developers/weaviate/client-libraries/python#batching"
```
Comment (dirkkul) [inline weaviate/error_msgs.py:18]: this is a v3 warning - I'd leave it like it is

File: weaviate/warnings.py [hunk 3]
```diff
@@ -81,8 +81,7 @@ def manual_batching() -> None:
         warnings.warn(
             message="""Dep002: You are batching manually. This means you are NOT using the client's built-in
             multi-threading. Setting `batch_size` in `client.batch.configure()`  to an int value will enabled automatic
-            batching. See:
-            https://weaviate.io/developers/weaviate/current/restful-api-references/batch.html#example-request-1""",
+            batching. See:https://weaviate.io/developers/weaviate/client-libraries/python#batching""",
```
Comment (dirkkul) [inline weaviate/warnings.py:84]: same, also old v3 warning

File: weaviate/warnings.py [hunk 4]
```diff
@@ -139,7 +138,7 @@ def use_of_client_batch_will_be_removed_in_next_major_release() -> None:
         warnings.warn(
             message="""Dep006: You are using the `client.batch()` method, which will be removed in the next major release.
             Please instead use the `client.batch.configure()` method to configure your batch and `client.batch` to enter the context manager.
-            See https://weaviate.io/developers/weaviate/client-libraries/python for details.""",
+            See https://weaviate.io/developers/weaviate/client-libraries/python#batching for details.""",
```
Comment (dirkkul) [inline weaviate/warnings.py:141]: same

File: CONTRIBUTING.md (hunk 1 above)
Comment (databyjp) [reply CONTRIBUTING.md:15]: My point is that I don't think client features get released without a corresponding server version. @dirkkul pls lmk if that's incorrect.

Comment (dirkkul) [reply CONTRIBUTING.md:15]: this is contributing, so for potential developers. I think it makes sense to have it in

Comment (databyjp) [reply CONTRIBUTING.md:15]: Then we should maybe point them towards running a dev setup directly.

**Review by daveatweaviate — COMMENTED** [review]
@dirkkul @databyjp thanks for the feedback, this is ready for another look

**Review by dirkkul — APPROVED** [review]
Thanks @daveatweaviate!

We merged the beta branch to main in the mean time, could you change the target? In case the PR is then borked: You can branch of main, cherry-pick, the commits from here and make a new PR, I'Ll merge that right away.

**Review by dirkkul — APPROVED** [review]

**Pull request closed without merging**